### PR TITLE
setup-environment-internal: cleanup variables

### PR DESCRIPTION
**Pull request by twoerner**
Several shell variables are leaked out to the user's shell when this script is
sourced. This patch cleans them up. 'BUILDDIR' is purposefully leaked because
it is used by 'devtool', for example.

Signed-off-by: Trevor Woerner <twoerner@gmail.com>

### CODE REVIEW

File: setup-environment-internal [hunk 1]
```diff
@@ -273,6 +280,10 @@ EOF
         fi
     fi
 fi
+unset EULA
```
Comment (ndechesne) [inline setup-environment-internal]: can you please put all of them on 1 line, e.g.

unset VAR1 VAR2 VAR3

File: setup-environment-internal [hunk 2]
```diff
@@ -297,3 +308,7 @@ Some of common targets are:
     core-image-minimal
 
 EOF
+
+unset MACHINE
```
Comment (ndechesne) [inline setup-environment-internal]: ditto

File: setup-environment-internal [hunk 3]
```diff
@@ -200,6 +203,10 @@ TMPDIR = "${OEROOT}/${_BUILDDIR}/tmp-${DISTRO_DIRNAME}"
 
 _EOF
 fi
+unset DISTRO_DIRNAME
+export BUILDDIR=$OEROOT/$_BUILDDIR
```
Comment (ndechesne) [inline setup-environment-internal]: i would prefer to do that in another commit. it's not a clean up, but a new feature (or bug fix)

Comment (ndechesne) [inline setup-environment-internal]: note that if we ever export BUILDDIR, we should get rid of _BUILDDIR all together. i am not sure why @koenkooi  used _BUILDDIR and not BUILDDIR in the first place. maybe he can comment.

**Comment by ndechesne** [comment]
also, using "declare -p" to check variables, i noticed that oldmach is left over, even with your patch.

**Comment by twoerner** [comment]
I've updated the branch and therefore the pull request. Please let me know what you think.

File: setup-environment-internal [hunk 4]
```diff
@@ -130,9 +130,9 @@ fi
 MANIFESTS="${OEROOT}"/.repo/manifests
 
 # we can be called with only 1 parameter max, <build> folder, or default to build-$distro
-_BUILDDIR=build-$DISTRO
+BUILDDIR=$OEROOT/build-$DISTRO
```
Comment (ndechesne) [inline setup-environment-internal]: this is not correct. before your change _BUILDDIR was a relative path here. now it's absolute. however in case the user specifies a build dir on the command line , BUILDDIR will be a relative path in the next if statement.

if you try

source setup-environment foobar

then the conf files will be wrong.

Comment (ndechesne) [inline setup-environment-internal]: it's probably safer to simply do s/_BUILDDIR/BUILDDIR/ and not change the logic of the code.

File: setup-environment-internal [hunk 5]
```diff
@@ -187,7 +187,7 @@ DL_DIR = "${OEROOT}/downloads"
 # Where to save shared state
 SSTATE_DIR = "${OEROOT}/sstate-cache"
 
-TMPDIR = "${OEROOT}/${_BUILDDIR}/tmp-${DISTRO_DIRNAME}"
+TMPDIR = "${BUILDDIR}/tmp-${DISTRO_DIRNAME}"
```
Comment (ndechesne) [inline setup-environment-internal]: same here, you changed the logic of the code, and TMPDIR can be will end up as 

TMPDIR = "foobar/tmp-rpb"

File: setup-environment-internal [hunk 6]
```diff
@@ -297,3 +291,7 @@ Some of common targets are:
     core-image-minimal
 
 EOF
+
+unset MACHINETABLE MACHLAYERS DISTROTABLE DISTROLAYERS DISTRO_DIRNAME OEROOT
```
Comment (ndechesne) [inline setup-environment-internal]: this commit is commit. there is no more leaked variables. thanks.

there is still a minor issue, since we have a couple of error paths which would result is returning earlier, in which case we might leak... but the issue was there before this commit anyways. I am ok to merge this commit as-is, and we can fix the error path with a proper exit trap later.

File: setup-environment-internal (hunk 4 above)
Comment (twoerner) [inline setup-environment-internal]: I think devtool needs BUILDDIR to be an absolute path, and _BUILDDIR was written to use a relative path, so the logic needs to change somewhat. I've pushed a new fix that addresses these issues and works both ways (without requiring the user to specify an absolute path), hopefully you like it.
